chebi-update/src/main/java/org/reactome/release/chebiupdate/ChebiUpdater.java

```diff
@@ -115,76 +150,107 @@ public void updateChebiReferenceMolecules() throws SQLException, Exception
 			List<DataItem> chebiFormulae = entity.getFormulae();
 
-			updateReferenceEntities(molecule, chebiName, instanceEdit);
 
-			// Now, check to see if we need to update the ReferenceMolecule itself.
 			String moleculeIdentifier = (String) molecule.getAttributeValue(ReactomeJavaConstants.identifier);
 			String moleculeName = (String) molecule.getAttributeValuesList(ReactomeJavaConstants.name).get(0);
 			String moleculeFormulae = (String) molecule.getAttributeValue(ReactomeJavaConstants.formula);
 
 			String prefix = "ReferenceMolecule (DB ID: " + molecule.getDBID() + " / ChEBI ID: " + moleculeIdentifier + ") has changes: ";
 
-			updateMoleculeIdentifier(molecule, chebiID, moleculeIdentifier, prefix);
-			updateMoleculeName(molecule, chebiName, moleculeName, prefix);
-			updateMoleculeFormula(molecule, chebiFormulae, moleculeFormulae, prefix);
-			// Update the display name.
-			InstanceDisplayNameGenerator.setDisplayName(molecule);
-			adaptor.updateInstanceAttribute(molecule, ReactomeJavaConstants._displayName);
-			
-			if (!testMode)
+			reportIfMoleculeIdentifierChanged(molecule, chebiID, moleculeIdentifier, prefix);
+			boolean nameUpdated = updateMoleculeName(molecule, chebiName, moleculeName, prefix);
+			boolean formulaUpdated = updateMoleculeFormula(molecule, chebiFormulae, moleculeFormulae, prefix);
+			// If the ChEBI name was updated, update the referenceEntities that refer to the molecule.
+			if (nameUpdated)
+			{
+				updateReferenceEntities(molecule, chebiName, instanceEdit);
+			}
+			// Now, check to see if we need to update the ReferenceMolecule itself.
+			if (nameUpdated || formulaUpdated)
 			{
 				addInstanceEditToExistingModifieds(instanceEdit, molecule);
-				
-				if (adaptor.supportsTransactions())
-				{
-					adaptor.commit();
-				}
+				// Update the display name.
+				InstanceDisplayNameGenerator.setDisplayName(molecule);
+				adaptor.updateInstanceAttribute(molecule, ReactomeJavaConstants._displayName);
 			}
 		}
+		if (!testMode)
+		{
+			adaptor.commit();
+		}
+		else
+		{
+			adaptor.rollback();
+		}
+		
 		logger.info("*** Formula-fill changes ***");
 		logger.info(this.formulaFillSB.toString());
 		logger.info("*** Formula update changes ***");
 		logger.info(this.formulaUpdateSB.toString());
-		logger.info("*** Name update changes ***");
-		logger.info(this.nameSB.toString());
-		logger.info("*** Identifier update changes ***");
-		logger.info(this.identifierSB.toString());
+		
+		refEntChangeLog.info("# DB_ID\tCreator\tAffected ReferenceEntity\tNew ChEBI Name\tUpdated list of all names");
+		// Print the referenceEntities that have changes, sorted by who created them.
+		for (GKInstance creator : this.referenceEntityChanges.keySet().stream().sorted(this.personComparator).collect(Collectors.toList()))
+		{
+			for (String message : this.referenceEntityChanges.get(creator))
+			{
+				refEntChangeLog.info("{}\t{}",creator.toString(), message);
+			}
+		}
```

Comment:
Can extract this as a private method called something like 'reportReferenceEntityChanges' similar to how there is 'reportIfMoleculeIdentifierChanged'

Reply:
That's true.  Personally, I just find it easier to read methods doing multiple things when the related sections of behaviour are encapsulated